radarr/tests/test_radarr.py

```diff
@@ -0,0 +1,87 @@
+import pytest
+from requests.exceptions import ConnectionError, HTTPError
+
+from datadog_checks.dev.utils import get_metadata_metrics
+from datadog_checks.radarr import RadarrCheck
+
+
+def test_check(aggregator, instance):
+    check = RadarrCheck('radarr', {}, [instance])
+
+    with pytest.raises(ConnectionError):
+        check.check(instance)
+        aggregator.assert_all_metrics_covered()
+        aggregator.assert_metrics_using_metadata(get_metadata_metrics())
```

Comment:
These two assertions here don't do much, since no metrics are collected nor asserted.

Instead, it would be more interesting if you added them inside a test where metric collection is successful, either by mocking what the endpoint returns or in an integration test against the actual service with the `dd_environment` fixture.

What do you think?

Reply:
Thanks for the review, I updated the PR and moved around some of the tests logic to assert metrics in a test where metrics are actually expected to be emitted, let me know what you think